// chat-window/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var buttonStyles = {
  border: '1px solid #eee',
  borderRadius: 3,
  backgroundColor: '#FFFFFF',
  padding: '3px 10px'
};

var nameStyles = {
  color: '#F00',
  marginRight: '1.5rem'
};

var Message = function Message(message) {
  return _react2.default.createElement(
    'p',
    null,
    _react2.default.createElement(
      'span',
      { style: (0, _extends3.default)({}, nameStyles) },
      message.name
    ),
    _react2.default.createElement(
      'span',
      null,
      message.text
    )
  );
};

var ChatWindow = function ChatWindow(_ref) {
  var messages = _ref.messages;
  var _ref$style = _ref.style;
  var style = _ref$style === undefined ? {} : _ref$style;
  return _react2.default.createElement(
    'div',
    {
      style: (0, _extends3.default)({}, buttonStyles, style)
    },
    messages.map(function (msg) {
      return _react2.default.createElement(
        'p',
        null,
        _react2.default.createElement(
          'span',
          { style: nameStyles },
          msg.name,
          ':'
        ),
        _react2.default.createElement(
          'span',
          null,
          msg.text
        )
      );
    })
  );
};

exports.default = ChatWindow;
